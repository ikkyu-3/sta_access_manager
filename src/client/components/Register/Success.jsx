// @flow
import type { RegisterSuccessProps } from "@types";

import * as React from "react";
import SingleButton from "@components/SingleButton";
import styles from "./style.css";

class Success extends React.Component<RegisterSuccessProps> {
  componentDidMount() {
    this.timeoutId = setTimeout(() => {
      this.props.onChangeStep();
      this.props.autoPageTransition();
    }, 5000);
  }

  componentWillUnmount() {
    clearTimeout(this.timeoutId);
  }

  timeoutId: TimeoutID;

  render() {
    const { onEntryButton } = this.props;
    const { main, textOnly, contentWithButton } = styles;
    return (
      <div className={main}>
        <div className={`${contentWithButton} ${textOnly}`}>登録しました!!</div>
        <SingleButton
          text="入室する"
          className="next"
          onButtonClick={onEntryButton}
        />
      </div>
    );
  }
}
export default Success;
