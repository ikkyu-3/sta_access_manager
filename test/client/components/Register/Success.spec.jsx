import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Success from "@components/Register/Success";

const fn = () => {};

const setTimeoutSpy = jest
  .spyOn(window, "setTimeout")
  .mockImplementation(() => 999);

const clearTimeoutSpy = jest
  .spyOn(window, "clearTimeout")
  .mockImplementation(() => {});

describe("Register/Success.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(
        <Success onChangeStep={fn} sutoPageTransition={fn} onEntryButton={fn} />
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    let failure;

    beforeAll(() => {
      failure = shallow(
        <Success onChangeStep={fn} sutoPageTransition={fn} onEntryButton={fn} />
      );
    });

    it("componentDidMountは、setTimeoutを実行する", () => {
      const [first, second] = setTimeoutSpy.mock.calls[0];
      expect(first).toBeInstanceOf(Function);
      expect(second).toBe(5000);
      expect(failure.instance().timeoutId).toBe(999);
    });

    it("componentWillUnmountは、clearTimeoutを実行する", () => {
      failure.instance().componentWillUnmount();
      const [first] = clearTimeoutSpy.mock.calls[0];
      expect(first).toBe(failure.instance().timeoutId);
    });
  });
});
