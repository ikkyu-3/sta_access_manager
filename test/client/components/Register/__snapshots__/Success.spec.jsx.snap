// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register/Success.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<div
  className="main"
>
  <div
    className="contentWithButton textOnly"
  >
    登録しました!!
  </div>
  <div
    className="singleButton"
  >
    <button
      className="next"
      onClick={[Function]}
    >
      入室する
    </button>
  </div>
</div>
`;
